provisioning/WFLY-21122-WildFly-Feature_pack-Family.adoc

```diff
@@ -0,0 +1,214 @@
+* `microprofile`: An implementation of the MicroProfile spec
+* `microprofile-7.1`: An implementation of the MicroProfile spec version 7.1.
+
+The set of criteria can evolve (new ones added, some removed) during the life time of the WildFly project.
+
+Some criteria could have a longer life (e.g: `jakarta-ee`) than some others (e.g.: `microprofile-7.1` that is bound to a specific version).
+
+The benefit of criteria is that feature-packs that depend on WildFly feature-packs criteria will have a clearer contract 
+than the today feature-pack producer name (that doesn't capture the criteria information).
+For example, a feature-pack expecting `jakarta-ee-10` and not compatible with `jakarta-ee-12` 
+would be expected to fail when provisioned with a future WildFly release.
+
+The criteria of a feature-pack can be inherited by the feature-packs that depend on it. Not all criteria are automatically inherited, a selection of 
+criteria to inherit can be operated (see `Implementation notes` fo details).
+
+Each feature-pack would expose the following set of criteria:
+
+* WildFly EE feature-pack: `jakarta-ee-deployment`, `jakarta-ee`, `jakarta-ee-10`
+* WildFly feature-pack: `microprofile`, `microprofile-7.1` + the following ones inherited from WildFly EE : `jakarta-ee-deployment`, `jakarta-ee`
+* WildFly Preview feature-pack: `jakarta-ee-deployment`, `jakarta-ee`, `jakarta-ee-11`, `microprofile`, `microprofile-7.1`
+
+=== Hard Requirements
+
+* Allow to provision WildFly or WildFly Preview with other feature-packs compatible with the defined criteria.
+* Suppress (or at least deprecate and remove in a next version) the extra Preview feature-packs that have been defined to work with WildFly Preview: 
+cloud Preview, datasources Preview and GRPC Preview.
+* No enforcement for extra feature-packs to benefit from this feature. Already released feature-pack could be provisioned with new WildFly feature-packs
+
+NOTE: That is outside of this proposal but WildFly and WildFly Preview feature-packs, 
+in order to be valid alternatives, must expose a common set of Galleon `features`, `packages` and `layers`.
+
+=== Implementation notes
+
+The changes in the WildFly project are:
+
+* Bump to Galleon 7.x and WildFly Galleon Plugins 8.1.x
+* Update the three `wildfly-feature-pack-build.xml` files to define their family and criteria.
+* Update the WildFly full `wildfly-feature-pack-build.xml` file to inherit criteria from its WildFly EE dependency.
+
+`wildfly-feature-pack-build.xml` extract:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="wildfly@maven(org.jboss.universe:community-universe):current">
+    <family name="wildfly">
+      <criteria name="microprofile"/>
+      <criteria name="microprofile-7.1"/>
+    </family>
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-deployment+jakarta-ee">
```

Comment:
The allowedFamily="wildfly:" introduces a new concept that should be described.

I think it's basically a namespace thing, yes? The 'jakarta-ee' criteria as defined by the WildFly project itself, as opposed to some outside party's thing.